### PR TITLE
fix: System Information Leak

### PR DESCRIPTION
**Pull request by JenniferMah**
Jira Ticket: https://issues.corp.twilio.com/browse/DI-1196

### CODE REVIEW

File: src/main/java/com/twilio/http/Request.java [hunk 1]
```diff
@@ -145,9 +145,9 @@ public URL constructURL() {
             URI uri = new URI(stringUri);
             return uri.toURL();
         } catch (final URISyntaxException e) {
-            throw new ApiException("Bad URI: " + stringUri, e);
```
Comment (eshanholtz) [inline src/main/java/com/twilio/http/Request.java:148]: nit: can these say the same thing?

Comment (JenniferMah) [reply src/main/java/com/twilio/http/Request.java:148]: Yup it should be updated now.

File: src/main/java/com/twilio/http/Request.java [hunk 2]
```diff
@@ -189,7 +189,7 @@ private String buildURL() {
             String credentials = parsedUrl.getUserInfo() != null ? parsedUrl.getUserInfo() + "@" : null;
             return joinIgnoreNull("", protocol, credentials, host, urlPort, encodedPath, query, ref);
         } catch (final MalformedURLException | UnsupportedEncodingException e) {
-            throw new ApiException("Bad URL: " + url, e);
+            throw new ApiException("Bad URL");
```
[the remark this answers is not in the document]
Comment (JenniferMah) [reply src/main/java/com/twilio/http/Request.java]: The error message should include the index or more information that indicates what is causing an error with the URL.

File: src/main/java/com/twilio/http/Request.java [hunk 3]
```diff
@@ -145,9 +144,9 @@ public URL constructURL() {
             URI uri = new URI(stringUri);
             return uri.toURL();
         } catch (final URISyntaxException e) {
-            throw new ApiException("Bad URI: " + stringUri, e);
+            throw new ApiException("Bad URI: " + e.getMessage());
```
Comment (shwetha-manvinkurke) [inline src/main/java/com/twilio/http/Request.java:147]: Why can't we log the `url`/`uri` as well?

Comment (JenniferMah) [reply src/main/java/com/twilio/http/Request.java:147]: We can but it's already included at the end of the error message and will look redundant. 
Current output: `Bad URI: Illegal character in authority at index 7: http://{`
Add URI/URL:`Bad URI: http://{ Illegal character in authority at index 7: http://{`

**Review by shwetha-manvinkurke — APPROVED** [review]

**Pull request merged**